src/OpenLoco/Vehicles/VehicleHead.cpp

```diff
@@ -2676,9 +2676,16 @@ namespace OpenLoco::Vehicles
 
                 break;
             }
+
+            bool stationHadPreviousCargo = cargoStats.quantity != 0;
             cargoStats.quantity = Math::Bound::add(cargoStats.quantity, cargo.qty);
             station->updateCargoDistribution();
-            cargoStats.enroute_age = std::max(cargoStats.enroute_age, cargo.numDays);
+            cargoStats.enrouteAge = Math::Bound::add(cargoStats.enrouteAge, cargo.numDays);
+            if (stationHadPreviousCargo)
+            {
+                cargoStats.enrouteAge *= 1 - static_cast<float>(cargo.qty) / cargoStats.quantity;
```

Comment:
never use floats in this code base. Instead multiply both sides by a constant and then divide out the constant.
e.g.

`((cargo.qty << 8) / cargoStats.quantity) >> 8` 
You'll need to have a think about if it needs higher accuracy and if it needs to be promoted to 64bit.

Reply:
Urgh just noticed what you are doing will need to be more complex than that.
Do:
```cpp
const auto multiplier = (1 << 16) - (cargo.qty << 16) / cargoStats.quantity;
cargoStats.enrouteAge = (cargoStats.enrouteAge * multiplier) >> 16
```

Reply:
I see, thanks! I've left a comment to make it easier to understand what's happening.